### DISCUSSION TITLE
[firebase-messaging-sw.js]: "activated and is stopped" after some period of time with `type: 'module'`

### DISCUSSION THREAD

**Discussion by eliw00d** [firebase/firebase-js-sdk#6025] [Q&A]
I followed https://firebase.google.com/docs/cloud-messaging/js/receive and am using type module. However, after some period of time, the service worker ends up with the "activated and is stopped" status and I no longer get notifications.

What is the best way to handle this? The documentation is lacking for this type of scenario.

**Comment by eliw00d** [comment]
Whenever it is stopped, I no longer see firebase-messaging-sw.js in Sources. If I try to manually click on start I get `Failed to load the script unexpectedly` in the console similar to [this](https://stackoverflow.com/questions/70732394/service-worker-errors-when-restarted-only-in-chrome-failed-to-load-the-script-u/70870597). I know that service workers have a life cycle but I thought updating them would restart them. Any thoughts?

Using Chrome, for reference.

I manually register the service worker and it looks like this:
```jsx
  const serviceWorkerRegistration = await navigator.serviceWorker.register(`/firebase-messaging-sw.js?${new URLSearchParams(firebaseConfig).toString()}`, {
    scope: '/firebase-cloud-messaging-push-scope',
    type: 'module',
  })
  
  try {
    await serviceWorkerRegistration.update()
  } catch (error) {
    /* it is non blocking and we don't care if it failed */
  }
```

**Comment by MorganTrudeau** [comment]
I'm seeing the same thing. Trying to find a solution.

**Reply by eliw00d** [reply]
I ended up not using `type: 'module'`, I think that was causing more issues than it was worth.

**Comment by eddie0mohammed** [comment]
Hello,
I'm experiencing the same problme,
the service worker is activated and running,
but after some time, it is automatically set to not active.

Did anyone find a solution to this ?

**Reply by eliw00d** [reply]
The service worker stopping doesn't impact whether it handles messages. I frequently see mine in a stopped state but it will still handle messages when they are received (and the app is in the background). However, I had a lot of issues with `type: module` in general so I stopped using it.
